hawkbit-simple-ui/src/main/java/org/eclipse/hawkbit/ui/simple/security/OAuth2TokenManager.java

```diff
@@ -10,73 +10,48 @@
 import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
 import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager;
 import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
 import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
-import org.springframework.security.oauth2.client.endpoint.OAuth2AccessTokenResponseClient;
-import org.springframework.security.oauth2.client.endpoint.OAuth2RefreshTokenGrantRequest;
-import org.springframework.security.oauth2.client.endpoint.RestClientRefreshTokenTokenResponseClient;
-import org.springframework.security.oauth2.client.registration.ClientRegistration;
-import org.springframework.security.oauth2.core.OAuth2AccessToken;
-import org.springframework.security.oauth2.core.OAuth2RefreshToken;
-import org.springframework.security.oauth2.core.endpoint.OAuth2AccessTokenResponse;
 import org.springframework.security.oauth2.core.oidc.user.DefaultOidcUser;
 import org.springframework.stereotype.Component;
 
-import java.util.Optional;
-
 @Component
 @ConditionalOnProperty(prefix = "hawkbit.server.security.oauth2.client", name = "enabled")
 public class OAuth2TokenManager {
 
     private final OAuth2AuthorizedClientService clientService;
     private final OAuth2AuthorizedClientManager clientManager;
-    private final OAuth2AccessTokenResponseClient<OAuth2RefreshTokenGrantRequest> tokenResponseClient;
 
     OAuth2TokenManager(
             final OAuth2AuthorizedClientService clientService,
             final OAuth2AuthorizedClientManager clientManager
     ) {
         this.clientService = clientService;
         this.clientManager = clientManager;
-        this.tokenResponseClient = new RestClientRefreshTokenTokenResponseClient();
     }
 
     public String getToken(final OAuth2AuthenticationToken authentication) {
-        return Optional.ofNullable(authorizedToken(authentication)).orElse(
-                ((DefaultOidcUser) authentication.getPrincipal()).getIdToken().getTokenValue()
-        );
-    }
-
-    /**
-     * Tries to refresh the id token if it is expired and adds it to the request.
-     */
-    private String authorizedToken(final OAuth2AuthenticationToken authentication) {
+        final String currentToken = ((DefaultOidcUser) authentication.getPrincipal()).getIdToken().getTokenValue();
         String registrationId = authentication.getAuthorizedClientRegistrationId();
-        OAuth2AuthorizeRequest request = OAuth2AuthorizeRequest.withClientRegistrationId(registrationId).principal(authentication).build();
 
         // This ensures that there is a client already, otherwise we won't be able to call the manager for authorization
         OAuth2AuthorizedClient authorizedClient = clientService.loadAuthorizedClient(registrationId, authentication.getName());
-        if (authorizedClient == null) return null;
+        if (authorizedClient == null) return currentToken;
```

Comment:
If client isn't found why return a token ?

Reply:
Hehe, this is the tricky parts of how simple-ui rights are working. It hijacks the authentication process, hence the initial step where it fetches the rights to create the authorities, the authorizedClient does not exist. 

That's why this OIDC process is much more complex than it should've been.

<img width="1614" height="665" alt="image" src="https://github.com/user-attachments/assets/0a30e4b0-21b5-4e12-b33b-f0057fce8cc6" />
<img width="1588" height="851" alt="image" src="https://github.com/user-attachments/assets/7989b9c2-fd79-4a58-8f16-7414021008da" />